feature/contributor/src/main/java/io/github/droidkaigi/confsched2020/contributor/ui/viewmodel/ContributorsViewModel.kt

```diff
@@ -1,11 +1,60 @@
 package io.github.droidkaigi.confsched2020.contributor.ui.viewmodel
 
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.asLiveData
+import androidx.lifecycle.liveData
+import io.github.droidkaigi.confsched2020.ext.combine
+import io.github.droidkaigi.confsched2020.ext.toAppError
+import io.github.droidkaigi.confsched2020.ext.toLoadingState
+import io.github.droidkaigi.confsched2020.model.AppError
+import io.github.droidkaigi.confsched2020.model.Contributor
+import io.github.droidkaigi.confsched2020.model.LoadState
 import io.github.droidkaigi.confsched2020.model.repository.ContributorRepository
+import java.lang.Exception
 import javax.inject.Inject
 
 class ContributorsViewModel @Inject constructor(
     private val contributorRepository: ContributorRepository
 ) : ViewModel() {
-    // TODO Implement
+
+    data class UiModel(
+        val isLoading: Boolean,
+        val error: AppError?,
+        val contributors: List<Contributor>
+    ) {
+        companion object {
+            val EMPTY = UiModel(false, null, emptyList())
+        }
+    }
+
+    private var contributors: List<Contributor> = emptyList()
+
+    private val contributorsLoadStateLiveData: LiveData<LoadState<List<Contributor>>> = liveData {
+        emitSource(
+            contributorRepository.contributorContents()
+                .toLoadingState()
+                .asLiveData()
+        )
+
+        try {
+            contributorRepository.refresh()
+        } catch (exception: Exception) {
+            // We can show contributors with cache
```

Comment:
This approach has a problem and I believe the same issue exists on other screens.

This is the sequence `contributorsLoadStateLiveData` emits when the screen opens for the first time with no previous cache.
UiModel(isLoading=true, error=null, contributors=[])
UiModel(isLoading=false, error=null, contributors=[])
UiModel(isLoading=false, error=null, contributors=[Contributor(id=1386930, name=takahirom, iconUrl=https://github.com/takahirom.png, profileUrl=https://github.com/takahirom, rankOrder=1) ...])

The second one dismisses the loading indicator because the repository emits an empty list when no cache exists. When the API call takes longer, there is a state where the screen shows nothing until the API call successfully completes. 

In my previous commits, I was emitting an error LiveData right here. I noticed that it causes another issue. When I have cache in the DB and open the screen with no network(or any errors fetching API response), the screen intermittently does not load data from cache. I'm guessing it's caused by some kinda racing condition in `emitSource`...?

I left it as is for now so that it can be tackled as a separate PR since it's an existing issue on other screens.